### PR TITLE
CEDS-1436 Replace Notification model with NotificationPresentationData

### PR DESCRIPTION
**Pull request by MaciejRewera**
Since this class is used for presentation only, it contains only
presentation-relevant fields. Backend should send Notification data
in the same format, decoupling presentation model from DB model.

### CODE REVIEW

File: app/models/NotificationPresentationData.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+package models
+
+import java.time.Instant
+
+import play.api.libs.json.Json
+
+final case class NotificationPresentationData(
```
Comment (ed0906) [inline app/models/NotificationPresentationData.scala]: Any chance we can remove the `Data` suffix? it doesn't add much value

File: app/views/movements.scala.html [hunk 2]
```diff
@@ -49,7 +49,7 @@
                     <td class="table-cell" id="submissionType-@submission.conversationId">@submission.submissionType</td>
                     <td class="table-cell" id="submissionAction-@submission.conversationId">@submission.submissionAction</td>
                     <td class="table-cell" id="dateUpdated-@submission.conversationId">
-                        @{notifications.headOption.map(n => n.dateTimeReceived).map(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm").withZone(ZoneId.systemDefault()).format(_))}</td>
+                        @{notifications.headOption.map(n => n.timestampReceived).map(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm").withZone(ZoneId.systemDefault()).format(_))}</td>
```
Comment (SteveSugden) [inline app/views/movements.scala.html]: (Not part of this PR change, but...) Would it not be useful to extract this date formatting logic somewhere so that it can be reused by other parts of the UI?

Comment (ed0906) [reply app/views/movements.scala.html]: Yeh Ideally we should pull it into a class which we can unit test and share across modules

Comment (ed0906) [reply app/views/movements.scala.html]: Also `n => n.timestampReceived` is the same as `_.timestampReceived`

File: app/models/NotificationPresentationData.scala (hunk 1 above)
Comment (MaciejRewera) [reply app/models/NotificationPresentationData.scala]: Done
